src/common-service/src/main/java/org/apache/kylin/rest/service/CustomUserServiceWithAccess.java

```diff
@@ -0,0 +1,122 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kylin.rest.service;
+
+import org.apache.kylin.common.KylinConfig;
+import org.apache.kylin.metadata.user.ManagedUser;
+import org.apache.kylin.metadata.user.NKylinUserManager;
+import org.apache.kylin.query.security.AccessDeniedException;
+import org.apache.kylin.rest.exception.InternalErrorException;
+import org.apache.kylin.rest.security.AccessProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.util.Assert;
+
+import java.io.IOException;
+import java.util.List;
+
+public class CustomUserServiceWithAccess extends OpenUserService{
+
+    private Logger logger = LoggerFactory.getLogger(CustomUserServiceWithAccess.class);
+
+    public static final String SUPER_ADMIN = "ADMIN";
+
+    private final AccessProvider accessProvider;
+
+    private final UserService userService;
+
+    public CustomUserServiceWithAccess(AccessProvider accessProvider, UserService userService) {
+        Assert.notNull(accessProvider, "The accessProvider should not be null.");
+        Assert.notNull(userService, "The userService should not be null.");
+        this.accessProvider = accessProvider;
+        this.userService = userService;
+    }
+
+    /**
+     * check whether the new user can access Kylin System
+     */
+    @Override
+    public void createUser(UserDetails user) {
+        accessProvider.checkAccessible(user.getUsername());//visit bdp hive or other corp inner service
+        userService.createUser(user);
+    }
+
+    @Override
+    public void updateUser(UserDetails user) {
+        userService.updateUser(user);
+    }
+
+    @Override
+    public boolean userExists(String userName){
+        boolean ifRegistered = userService.userExists(userName);
+        if(!ifRegistered){
+            try{
+                accessProvider.checkAccessible(userName);// default true
+            } catch (AccessDeniedException e){
+                logger.info("user {} is not able to access Kylin due to ", userName, e);
+                return false;
+            }
+            ManagedUser newUser = new ManagedUser(userName);
+            userService.createUser(newUser);
+        }
+        return true;
```

Comment:
> 

this logic first check if the user in system just use KylinUserService to check, if false then use accessProvider.checkAccessible(userName) to check, if false it will return fasle then method loadUserByUserName in class SelfManagedAuthoritiesPopulator  would throw exception; if accessProvider.checkAccessible(userName) is true means user can access system and create user info.